src/__tests__/WeekView.test.js

```diff
@@ -1,10 +1,454 @@
+      expect(eventArg.id).toEqual(targetId);
+    });
+  });
+
+  describe('Dragging events', () => {
+    const startDate = new Date(2022, 3, 5, 12, 0, 0);
+    const endDate = new Date(2022, 3, 5, 14, 53, 52);
+
+    const targetId = 171;
+    const events = [
+      {
+        id: targetId,
+        startDate,
+        endDate,
+        description: 'event to be dragged',
+      },
+      {
+        id: 82,
+        startDate: new Date(2022, 3, 6, 17),
+        endDate: new Date(2022, 3, 6, 18),
+      },
+    ];
+
+    const numberOfDays = 3;
+
+    const mockAndRender = () => {
+      const mockOnDragEvent = jest.fn();
+
+      render(
+        <WeekView
+          events={events}
+          numberOfDays={numberOfDays}
+          selectedDate={new Date(2022, 3, 4)}
+          startHour={10}
+          hoursInDisplay={24}
+          onDragEvent={mockOnDragEvent}
+        />,
+      );
+
+      return mockOnDragEvent;
+    };
+
+    it('callback is not called if the movement fails', () => {
+      const mockOnDragEvent = mockAndRender();
+
+      fireGestureHandler(getByGestureTestId(`dragGesture-${targetId}`), [
+        { translationX: 100, translationY: -3 },
+        { state: State.FAILED },
+      ]);
+
+      expect(mockOnDragEvent).not.toHaveBeenCalled();
+    });
+
+    it('callback is not called if the movement is cancelled', () => {
+      const mockOnDragEvent = mockAndRender();
+
+      fireGestureHandler(getByGestureTestId(`dragGesture-${targetId}`), [
+        { translationX: -100, translationY: 200 },
+        { state: State.CANCELLED },
+      ]);
+
+      expect(mockOnDragEvent).not.toHaveBeenCalled();
+    });
+
+    it('callback is called with the correct arguments', () => {
+      const mockOnDragEvent = mockAndRender();
+
+      fireGestureHandler(getByGestureTestId(`dragGesture-${targetId}`), [
+        { translationX: 100, translationY: 100 },
+      ]);
+
+      expect(mockOnDragEvent).toHaveBeenCalledTimes(1);
+      const [e, newStartDate, newEndDate] = mockOnDragEvent.mock.calls[0];
+
+      expect(e.id).toEqual(targetId);
+      expect(newStartDate).toBeValidDate();
+      expect(newEndDate).toBeValidDate();
+    });
+
+    describe('when dragging in direction', () => {
+      const startOfDay = new Date(startDate.getTime());
+      startOfDay.setHours(0);
+      startOfDay.setMinutes(0);
+      startOfDay.setSeconds(0);
+      const endOfDay = new Date(startDate.getTime());
+      endOfDay.setHours(23);
+      endOfDay.setMinutes(59);
+      endOfDay.setSeconds(59);
+
+      /**
+       * NOTE: the screen-width (in pixels) is needed to simulate a drag gesture
+       * through a specific amount of days.
+       */
+      const estimatedDayWidth = Dimensions.get('window').width / numberOfDays;
+      const atLeastOneDay = Math.floor(estimatedDayWidth * 1.5);
+
+      if (Number.isNaN(estimatedDayWidth)) {
+        throw new Error('Could not get window width');
+      }
```

Comment:
the test should not have if case, can we avoid it?

Reply:
> can we avoid it?

yeah, I'll remove it. Apparently dimensions in jest are constant: https://github.com/facebook/react-native/blob/0.68-stable/jest/setup.js#L213 (could not find docs about it, though)